### PR TITLE
[Feature] Added Mongo Storage and cache

### PR DESCRIPTION
**Pull request by PascalDR**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+      MYSQL_DATABASE: '${MYSQL_DATABASE}'
+      MYSQL_USER: '${MYSQL_USER}'
+      MYSQL_PASSWORD: '${MYSQL_PASSWORD}'
+      MYSQL_ROOT_PASSWORD: '${MYSQL_ROOT_PASSWORD}'
+    volumes:
+      - db-data:/var/lib/mysql
+    networks:
+      - wordpress-network
+
+  phpmyadmin:
```
Comment (peppelinux) [inline docker-compose.yml:21]: please comment this to leave it at the will of the user having phpadmin or not

File: example/satosa/pyeudiw_backend.yaml [hunk 2]
```diff
@@ -71,6 +73,8 @@ config:
           use: sig
           x5c:
           - '...'
+      # time in ms
+      token_exp_delta: 600000
```
Comment (peppelinux) [inline example/satosa/pyeudiw_backend.yaml:77]: this collides with the new configuration, please use it

File: pyeudiw/satosa/backend.py [hunk 3]
```diff
@@ -32,18 +33,23 @@ def __init__(self, auth_callback_func, internal_attributes, config, base_url, na
...
+  
         self.client_id = config['wallet_relying_party']['client_id']
+  
+        self.default_sign_alg = config['default_sign_alg']
         
         self.absolute_redirect_url = config['wallet_relying_party']['redirect_uris'][0]
         self.absolute_request_url = config['wallet_relying_party']['request_uris'][0]
 
-        self.qrcode_settings = config['qrcode_settings']
+        self.qr_settings = config['qr_code_settings']
```
Comment (peppelinux) [inline pyeudiw/satosa/backend.py:50]: this has changed in dev branch, please align this accordingly

File: pyeudiw/satosa/backend.py [hunk 4]
```diff
@@ -32,18 +33,23 @@ def __init__(self, auth_callback_func, internal_attributes, config, base_url, na
...
+        self.token_exp_delta = config['jwks']['token_exp_delta']
```
Comment (peppelinux) [inline pyeudiw/satosa/backend.py:51]: the same comment as the previous

**Review by peppelinux — CHANGES_REQUESTED** [review]

**Review by peppelinux — CHANGES_REQUESTED** [review]

File: pyeudiw/storage/mongo_cache.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        self.url = url
+        self.connection_params = connection_params
+
+        self.client = None
+        self.db = None
+        
+    def _connect(self):
+        if not self.client or not self.client.server_info():
+            self.client = pymongo.MongoClient(self.url, **self.connection_params)
+            self.db = getattr(self.client, self.storage_conf["db_name"])
```
[the remark this answers is not in the document]
Comment (PascalDR) [reply pyeudiw/storage/mongo_cache.py:21]: Resolved in commit bbcd929b20eea5034251a65fbbcb7fd5f75d18a4

**Review by peppelinux — APPROVED** [review]

**Pull request merged**